.github/workflows/diffcalc.yml

```diff
@@ -331,17 +338,21 @@ jobs:
           cd "${{ needs.directory.outputs.GENERATOR_DIR }}"
           docker-compose down
 
+  output-cli:
+    name: Output info
+    needs: generator
+    runs-on: ubuntu-latest
+    steps:
       - name: Output info
-        if: ${{ success() }}
         run: |
-          echo "Target: ${{ steps.run.outputs.TARGET }}"
-          echo "Spreadsheet: ${{ steps.run.outputs.SPREADSHEET_LINK }}"
+          echo "Target: ${{ needs.generator.outputs.TARGET }}"
+          echo "Spreadsheet: ${{ needs.generator.outputs.SPREADSHEET_LINK }}"
 
   update-comment:
     name: Update PR comment
     needs: [ create-comment, generator ]
     runs-on: ubuntu-latest
-    if: ${{ github.event_name == 'issue_comment' && github.event.issue.pull_request && contains(github.event.comment.body, '!diffcalc') && github.event.comment.author_association == 'OWNER' }}
+    if: ${{ always() && needs.create-comment.result == 'success' }}
```

Comment:
This is a bit convoluted...
- Job specifies two `needs`, which means this job won't execute unless the two needed jobs pass...
- ...except there's the `always()` in there, which means the job will always execute...
- ...unless the `create-comment` job will fail, in which case it won't.

I'm suspecting that the only goal here is to establish a _temporal_ dependency rather than an _output_ dependency, i.e. ensure that `update-comment` does not run without either `create-comment` and `generator` completing first, regardless of result? Would that be correct understanding?

Reply:
Yeah okay that more or less confirms my initial understanding of it.